spark/v3.2/spark-extensions/src/main/scala/org/apache/spark/sql/execution/dynamicpruning/RowLevelCommandDynamicPruning.scala

```diff
@@ -109,6 +123,18 @@ case class RowLevelCommandDynamicPruning(spark: SparkSession) extends Rule[Logic
     dynamicPruningSubqueries.reduce(And)
   }
 
+  private def buildAttrMap(
+      tableAttrs: Seq[Attribute],
+      scanAttrs: Seq[Attribute]): AttributeMap[Attribute] = {
+
+    val attrMapping = tableAttrs.flatMap { tableAttr =>
+      scanAttrs
+        .find(scanAttr => scanAttr.name == tableAttr.name)
```

Comment:
Should this use the resolver or are we confident that the case will always match?

Reply:
We could use a resolver to be safe but we are sure the names will match as the other attribute is constructed by cloning the original one. I'll update anyway.

Reply:
Switched to using a resolver.